src/main.rs

```diff
@@ -46,26 +46,14 @@ Use MK_LOG=trace to see debug output.
         );
         exit(0);
     }
-    match Newer::new(args) {
-        Ok(newer) => {
-            if !newer.should_run_command() {
-                debug!("Nothing to do.");
-                exit(if newer.newer { 1 } else { 0 });
-            } else {
-                match run_command(newer.command) {
-                    Ok(_) => {}
-                    Err(e) => {
-                        eprintln!("{}", e);
-                        exit(1);
-                    }
-                }
-            }
-        }
-        Err(e) => {
-            eprintln!("{}", e);
-            exit(1);
-        }
-    }
+    let newer = Newer::new(args)?;
+    let code = if newer.should_run_command() {
+        run_command(newer.command)?
+    } else {
+        debug!("Nothing to do.");
+        newer.newer.into()
```

Comment:
I believe this does the same as `if newer.newer { 1 } else { 0 }`…

Reply:
Ah neat.